src/dsl.ts

```diff
@@ -254,40 +259,26 @@ export class SnykApiCheckDsl implements ApiCheckDsl {
   }
 
   get specification() {
-    const change: IChange = {
-      location: {
-        conceptualLocation: { path: "This Specification", method: "" },
-        jsonPath: "/",
-        conceptualPath: [],
-        kind: "API",
-      },
-    } as any;
-
-    const value: ShouldOrMust<
-      (
-        document: OpenAPIV3.Document,
-        context: SynkApiCheckContext,
-        docs: DocsLinkHelper,
-      ) => Promise<void> | void
-    > = {
-      must: (statement, handler) => {
-        const docsHelper = newDocsLinkHelper();
-        this.checks.push(
-          runCheck(
-            change,
-            docsHelper,
-            "this specification: ",
-            statement,
-            true,
-            () => handler(this.nextJsonLike, this.providedContext, docsHelper),
-          ),
-        );
-      },
-    };
+    const dsl = this;
 
-    return {
-      requirement: value,
-    };
+    return genericEntityRuleImpl<
```

Comment:
We now model specifications as a "Fact" which means we can use the generic rule implementation. In the future, we're going to move this into optic internals so the consuming code doesn't need to contain implementation details.

The effect of this change is some _slightly_ different error messages - see snapshot changes for details (`this specification` -> `requirement for specification`)